src/OpenTelemetry.Instrumentation.Dependencies/HttpClientInstrumentationOptions.cs

```diff
@@ -54,54 +35,55 @@ internal HttpClientInstrumentationOptions(Func<string, object, object, bool> eve
+        /// Gets or sets an optional callback method for filtering HttpClient requests that are sent through the instrumentation.
         /// </summary>
-        internal Func<string, object, object, bool> EventFilter { get; set; }
+        public Func<HttpRequestMessage, bool> FilterFunc { get; set; }
 
-        private static bool DefaultFilter(string activityName, object arg1, object unused)
+        internal static bool IsInternalUrl(Uri requestUri)
         {
-            // TODO: there is some preliminary consensus that we should introduce 'terminal' spans or context.
-            // exporters should ensure they set it
+            var originalString = requestUri.OriginalString;
 
-            if (IsHttpOutgoingPostRequest(activityName, arg1, out Uri requestUri))
+            // zipkin
+            if (originalString.Contains(":9411/api/v2/spans"))
             {
-                var originalString = requestUri.OriginalString;
-
-                // zipkin
-                if (originalString.Contains(":9411/api/v2/spans"))
-                {
-                    return false;
-                }
+                return true;
+            }
 
-                // applicationinsights
-                if (originalString.StartsWith("https://dc.services.visualstudio") ||
-                    originalString.StartsWith("https://rt.services.visualstudio") ||
-                    originalString.StartsWith("https://dc.applicationinsights") ||
-                    originalString.StartsWith("https://live.applicationinsights") ||
-                    originalString.StartsWith("https://quickpulse.applicationinsights"))
-                {
-                    return false;
-                }
+            // applicationinsights
+            if (originalString.StartsWith("https://dc.services.visualstudio") ||
+                originalString.StartsWith("https://rt.services.visualstudio") ||
+                originalString.StartsWith("https://dc.applicationinsights") ||
+                originalString.StartsWith("https://live.applicationinsights") ||
+                originalString.StartsWith("https://quickpulse.applicationinsights"))
+            {
+                return true;
             }
 
-            return true;
+            return false;
         }
 
-        private static bool IsHttpOutgoingPostRequest(string activityName, object arg1, out Uri requestUri)
+        internal bool EventFilter(string activityName, object arg1)
         {
-            if (activityName == "System.Net.Http.HttpRequestOut")
+            if (TryParseHttpRequestMessage(activityName, arg1, out HttpRequestMessage requestMessage))
             {
-                if (arg1 is HttpRequestMessage request &&
-                    request.RequestUri != null &&
-                    request.Method == HttpMethod.Post)
+                Uri requestUri;
+                if (requestMessage.Method == HttpMethod.Post && (requestUri = requestMessage.RequestUri) != null)
```

Comment:
do we have to filter **only** Post requests?

Reply:
That is only for the internal filter, which happens to be all POSTs. The user filter will fire for all methods.

Reply:
There were GET requests made by application insights exporter as well. (though only once at startup).
Now that we expose public api to filter, do we still need to have zipkin specific one?

Reply:
* Interesting, didn't know about the Azure GETs. The "POST" rule has been part of the filter since I have been working on it, so I'm not sure why it was originally put in. You want me to open up to all methods? Are the GET URLs the same as the POST ones?

* I think we still need the zipkin automatic filter unless you are thinking we can direct users to manually add that filter via the public API if they decide to use the ZipkinExporter?

   The ZipkinExporter in-theory could use the same public API. But it would have to somehow find the instance of the options, which is in the Dependencies assembly it doesn't have a reference to. Then it would have to co-op the user's filter, if they added one. It could get messy. Prefer leaving as-is and solving in a generic way. The whole "terminal span" thing.

Reply:
Yes agree to leave this for now.